queue/group_local.go

```diff
@@ -70,16 +117,26 @@ func (g *localQueueGroup) Queues(_ context.Context) []string {
 	return g.cache.Names()
 }
 
-// Get a queue with the given index. Get sets the last accessed time to now. Note that this means
-// that the caller must add a job to the queue within the TTL, or else it may have attempted to add
-// a job to a closed queue.
-func (g *localQueueGroup) Get(ctx context.Context, id string) (amboy.Queue, error) {
+// Get a queue with the given id. Get sets the last accessed time to now. Note
+// that this means that the time between when the queue is retrieved and when
+// the caller actually performs an operation on the queue (e.g. add a job) must
+// be within the TTL; otherwise, the queue might be closed before the operation
+// is done.
+func (g *localQueueGroup) Get(ctx context.Context, id string, opts ...amboy.QueueOptions) (amboy.Queue, error) {
 	q := g.cache.Get(id)
 	if q != nil {
 		return q, nil
 	}
 
-	queue, err := g.opts.Constructor(ctx)
+	localQueueOpts, err := getLocalQueueOptions(opts...)
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid queue options")
+	}
+	queueOpts := mergeLocalQueueOptions(append([]LocalQueueOptions{g.opts.Queue}, localQueueOpts...)...)
```

Comment:
Should the passed in opts take precedence here?

Reply:
Haha how embarrassing, I read `mergeLocalQueueOptions` incorrectly. I thought it was only setting it if it wasn't already set.